### PR TITLE
BigQuery to Google Ads Connector

### PR DESCRIPTION
**Pull request by blackwire**
Adds a new connector that provides a way to take values (associated with GCLIDs) from a BigQuery table and import them into Google Ads as Offline Click Conversions.

### CODE REVIEW

File: backend/jobs/workers/bigquery/bq_to_ads_offline_click_conversion.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+class AdsOfflineClickPageResultsWorker(bq_batch_worker.TablePageResultsProcessorWorker):
+  """A page results worker for uploading a chunk of conversion data."""
+
+  def _process_page_results(self, page_data: page_iterator.Page) -> None:
+    ads_client = self._get_ads_client()
+    num_rows = page_data.num_items
+    template = string.Template(self._params['template'])
+
+    conversion_objs = []
```
Comment (blackwire) [inline backend/jobs/workers/bigquery/bq_to_ads_offline_click_conversion.py:158]: The limit here is 2000 per request. So, we probably want to have a failsafe here to ensure we don't go over that number.

Comment (blackwire) [reply backend/jobs/workers/bigquery/bq_to_ads_offline_click_conversion.py:158]: https://developers.google.com/google-ads/api/docs/best-practices/quotas#:~:text=Limited%20to%202%2C000%20call%20or%20click%20conversions%20per%20request

File: backend/jobs/workers/bigquery/bq_batch_worker.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      page_size=self.DEFAULT_BQ_BATCH_SIZE
+    )
+
+    enqueued_jobs_count = 0
+    for _ in row_iterator.pages:
+
+      # Enqueue job for this page
+      worker_params = self._params.copy()
+      worker_params[BQ_PAGE_TOKEN_PARAM] = page_token
+      worker_params[BQ_BATCH_SIZE_PARAM] = self.DEFAULT_BQ_BATCH_SIZE
```
Comment (blackwire) [inline backend/jobs/workers/bigquery/bq_batch_worker.py:83]: It seems "bq_batch_size" param is ignored for now. Perhaps we should implement this as an actual default.

**Review by jmistral-google — APPROVED** [review]
Looks great, thanks for making all of these improvements.

**Pull request merged**